### PR TITLE
Fix error due to missing VDE on Linux

### LINKED ISSUE #1031: Error starting VM due to missing `vde` on Linux
### Rancher Desktop Version

Commit dc578f9 (on main branch)

### Rancher Desktop K8s Version

N/A

### What operating system are you using?

Other Linux

### Operating System / Build Version

Pop OS

### What CPU architecture are you using?

x64

### Windows User Only

_No response_

### Actual Behavior

Running `npm run dev` on the `main` branch results in a failure to start the VM.

### Steps to Reproduce

1. Start Rancher Desktop with `npm run dev`
2. Application should fail when starting VM

### Result

The VM fails to start with the message as shown in the screenshot below.

### Expected Behavior

Normal startup.

### Additional Information

image

`vde` is associated with MacOS; we should not be getting references to it on Linux.

### PR DESCRIPTION
**Pull request by adamkpickering**
Fixes #1031.

I tested this on Linux via `npm run dev` and it appears to work fine.

Not tested on MacOS. It should be, but I don't have the equipment required.

### CODE REVIEW

File: src/assets/lima-config.yaml [hunk 1]
```diff
@@ -48,13 +48,15 @@ provision:
...
-    ETH0="$(ip route | grep "^default.* dev eth0 ")"
-    RD0="$(ip route | grep "^default.* dev rd0 ")"
-    IP="$(echo "$RD0" | awk '{print $3}')"
-    METRIC="$(echo "$ETH0" | awk '{print $NF}')"
-    ip route del default via "$IP" dev rd0
-    ip route add default via "$IP" dev rd0 metric $((METRIC - 1))
+    if [ "$(uname)" == 'Darwin' ]; then
+      set -o errexit -o nounset -o xtrace
+      ETH0="$(ip route | grep "^default.* dev eth0 ")"
+      RD0="$(ip route | grep "^default.* dev rd0 ")"
```
Comment (jandubois) [inline src/assets/lima-config.yaml:54]: Sorry, I know I suggested the `uname` test, but it is not going to work: the provisioning script runs inside the VM, so will always be Linux.

Instead we can check if the `rd0` interface exists, and only try to manipulate it's metric when it does:

```shell
if [ -n "${RD0}" ]; then
```

File: src/k8s-engine/lima.ts [hunk 2]
```diff
@@ -455,7 +455,7 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
   protected async updateConfig(desiredVersion: semver.SemVer) {
     const currentConfig = await this.currentConfig;
     const baseConfig: Partial<LimaConfiguration> = currentConfig || {};
-    const config: LimaConfiguration = merge({}, baseConfig, DEFAULT_CONFIG as LimaConfiguration, {
```
[the remark this answers is not in the document]
Comment (adamkpickering) [reply src/k8s-engine/lima.ts:458]: I agree that it's a good idea to not modify baseConfig, also that a comment would be helpful here. Done.

File: src/assets/lima-config.yaml (hunk 1 above)
Comment (adamkpickering) [reply src/assets/lima-config.yaml]: Done

**Review by jandubois — APPROVED** [review]
I've not tested the changes, but the code looks fine to me now.

**Comment by jandubois** [comment]
> Not tested on MacOS.

I've now tested it on macOS, and it worked fine.

**Pull request merged**